### PR TITLE
[DEV-1962] implement urlreplacemap

### PR DESCRIPTION
**Pull request by MarBert**
#### List of Changes

This PR merges static UrlReplacesMap with the ones from strapi
Depends on PR [#1166](https://github.com/pagopa/developer-portal/pull/1166/)
#### How Has This Been Tested?

On localhost, i used the documentation "aBQM7h48Vhhg8le4VIK8", where a url to the old doc is used. 
Remove `'https://docs.pagopa.it/kb-enti': '/app-io/guides/supporto-agli-enti',` from apps/nextjs-website/src/_contents/products.ts and add the entry on strapi to test it
[link to guide with url ](http://localhost:3000/app-io/guides/io-guida-tecnica)
#### Screenshots (if appropriate):
image
image

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

[DEV-1951]: https://pagopa.atlassian.net/browse/DEV-1951?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/nextjs-website/src/app/solutions/[solutionSlug]/[...solutionSubPathSlugs]/page.tsx [hunk 1]
```diff
@@ -64,6 +60,7 @@ const Page = async ({ params }: { params: Params }) => {
     params?.solutionSubPathSlugs
   );
 
+  const urlReplacesMap = (await getUrlReplaceMapProps()) || {};
```
Comment (tommaso1) [inline apps/nextjs-website/src/app/solutions/[solutionSlug]/[...solutionSubPathSlugs]/page.tsx:63]: it seems it's declared but not used

File: apps/nextjs-website/src/lib/cmsApi.ts [hunk 2]
```diff
@@ -136,6 +139,17 @@ export const getQuickStartGuidesProps = async () => {
...
 };
 
+export const getUrlReplaceMapProps = async () => {
+  const {
+    config: { FETCH_FROM_STRAPI: fetchFromStrapi },
+  } = buildEnv;
+
+  if (fetchFromStrapi) {
+    const urlReplaceMap = await fetchUrlReplaceMap(buildEnv);
+    return makeUrlReplaceMap(urlReplaceMap);
```
Comment (tommaso1) [inline apps/nextjs-website/src/lib/cmsApi.ts:149]: maybe in here it should be done the merge with the static ones

File: apps/nextjs-website/src/app/solutions/[solutionSlug]/[...solutionSubPathSlugs]/page.tsx (hunk 1 above)
Comment (MarBert) [reply apps/nextjs-website/src/app/solutions/[solutionSlug]/[...solutionSubPathSlugs]/page.tsx:63]: it's used later, line 79, to build the props ( i just changed the declaration with the new map)

File: apps/nextjs-website/src/lib/cmsApi.ts [hunk 3]
```diff
@@ -136,6 +140,21 @@ export const getQuickStartGuidesProps = async () => {
...
+  } = buildEnv;
+
+  if (fetchFromStrapi) {
+    const urlReplaceMap = await fetchUrlReplaceMap(buildEnv);
+    return Object.assign(
+      {},
+      staticUrlReplacesMap,
+      makeUrlReplaceMap(urlReplaceMap)
+    );
+  }
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/cmsApi.ts:155]: I think that, for now, you should return staticUrlReplaceMap if `fetchFromStrapi`  is `false`

File: apps/nextjs-website/src/lib/strapi/fetches/fetchUrlReplaceMap.ts [hunk 4]
```diff
@@ -4,17 +4,25 @@ import qs from 'qs';
...
+              'bannerLinks.icon',
+              'seo',
+              'seo.metaSocial.image',
+              'product.logo',
+              'product.bannerLinks.icon',
+            ],
+          },
+        },
+      },
+    },
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/fetches/fetchUrlReplaceMap.ts:25]: :+1:

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/app/page.tsx [hunk 5]
```diff
@@ -8,7 +8,7 @@ import {
 } from '@/helpers/metadata.helpers';
 import dynamic from 'next/dynamic';
 import { baseUrl } from '@/config';
-import { getHomepageProps } from '@/lib/cmsApi';
+import { getHomepageProps, getUrlReplaceMapProps } from '@/lib/cmsApi';
```
Comment (marcobottaro) [inline apps/nextjs-website/src/app/page.tsx:11]:
```suggestion
import { getHomepageProps } from '@/lib/cmsApi';
```

File: apps/nextjs-website/src/lib/cmsApi.ts [hunk 6]
```diff
@@ -136,6 +140,21 @@ export const getQuickStartGuidesProps = async () => {
...
+  } = buildEnv;
+
+  if (fetchFromStrapi) {
+    const urlReplaceMap = await fetchUrlReplaceMap(buildEnv);
+    return Object.assign(
+      {},
+      staticUrlReplaceMap,
+      makeUrlReplaceMap(urlReplaceMap)
+    );
+  } else return staticUrlReplaceMap;
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/cmsApi.ts:155]:
```suggestion
  }
  return staticUrlReplaceMap;
```

File: apps/nextjs-website/src/app/[productSlug]/guides/[...productGuidePage]/page.tsx [hunk 7]
```diff
@@ -81,7 +77,7 @@ const Page = async ({ params }: { params: Params }) => {
     params?.productSlug,
     params?.productGuidePage ?? ['']
   );
-
+  const urlReplaceMap = (await getUrlReplaceMapProps()) || {};
```
Comment (marcobottaro) [inline apps/nextjs-website/src/app/[productSlug]/guides/[...productGuidePage]/page.tsx:80]: I think that ` || {}` is not necessary if you return the staticUrlReplaceMap in getUrlReplaceMapProps function
```suggestion
  const urlReplaceMap = await getUrlReplaceMapProps();
```

File: apps/nextjs-website/src/app/[productSlug]/tutorials/[...productTutorialPage]/page.tsx [hunk 8]
```diff
@@ -124,6 +121,7 @@ const Page = async ({ params }: { params: Params }) => {
     );
   }
 
+  const urlReplaceMap = (await getUrlReplaceMapProps()) || {};
```
Comment (marcobottaro) [inline apps/nextjs-website/src/app/[productSlug]/tutorials/[...productTutorialPage]/page.tsx:124]: same here

File: apps/nextjs-website/src/app/solutions/[solutionSlug]/[...solutionSubPathSlugs]/page.tsx [hunk 9]
```diff
@@ -64,6 +60,7 @@ const Page = async ({ params }: { params: Params }) => {
     params?.solutionSubPathSlugs
   );
 
+  const urlReplaceMap = (await getUrlReplaceMapProps()) || {};
```
Comment (marcobottaro) [inline apps/nextjs-website/src/app/solutions/[solutionSlug]/[...solutionSubPathSlugs]/page.tsx:63]: same here

**Review by marcobottaro — CHANGES_REQUESTED** [review]

**Review by marcobottaro — APPROVED** [review]

File: apps/nextjs-website/src/lib/cmsApi.ts [hunk 10]
```diff
@@ -136,6 +140,22 @@ export const getQuickStartGuidesProps = async () => {
...
+    config: { FETCH_FROM_STRAPI: fetchFromStrapi },
+  } = buildEnv;
+
+  if (fetchFromStrapi) {
+    const urlReplaceMap = await fetchUrlReplaceMap(buildEnv);
+    return Object.assign(
+      {},
+      staticUrlReplaceMap,
+      makeUrlReplaceMap(urlReplaceMap)
+    );
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/cmsApi.ts:154]:
```suggestion
    return { ...staticUrlReplaceMap, ...makeUrlReplaceMap(urlReplaceMap) };
```

**Review by marcobottaro — CHANGES_REQUESTED** [review]

**Review by marcobottaro — APPROVED** [review]

**Pull request merged**